### PR TITLE
Adding popup for FIPs Search in MadLib

### PR DESCRIPTION
**Pull request by kkatzen**
#151

### CODE REVIEW

File: app/src/pages/ui/FipsSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  return (
+    <>
+      <Button
+        variant="text"
+        className={styles.MadLibButton}
+        onClick={handleClick}
+      >
+        {new Fips(props.value).getFullDisplayName()}
+        {!open && <ArrowDropDown />}
+        {open && <ArrowDropUp />}
```
Comment (aaronsn) [inline app/src/pages/ui/FipsSelector.tsx:39]: nit: I'd do a ternary operator here instead

File: app/src/pages/ui/FipsSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                {...params}
+              />
+            )}
+            onChange={(e, fips) => {
+              props.onGeoUpdate(fips.code);
+              handleClose();
+            }}
+          />
+          <span className={styles.NoteText}>
+            Note: City and census tract location is currently unavailable
```
Comment (aaronsn) [inline app/src/pages/ui/FipsSelector.tsx:77]: nit: drop "Note: "

File: app/src/pages/ui/FipsSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          <Autocomplete
+            disableClearable={true}
+            options={props.options}
+            clearOnEscape={true}
+            getOptionLabel={(fips) => fips.getFullDisplayName()}
+            getOptionSelected={(fips) => fips.code === props.value}
+            renderOption={(fips) => <>{fips.getFullDisplayName()}</>}
+            renderInput={(params) => (
+              <TextField
+                placeholder="County, State, Territory or United States" // TODO- update depending on what options are
```
Comment (aaronsn) [inline app/src/pages/ui/FipsSelector.tsx:65]: nit: add comma after "Territory"

File: app/src/pages/ui/FipsSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import { Fips } from "../../utils/madlib/Fips";
+import Popover from "@material-ui/core/Popover";
+import Button from "@material-ui/core/Button";
+import styles from "./FipsSelector.module.scss";
+
+function FipsSelector(props: {
+  value: string;
+  options: Fips[];
+  onGeoUpdate: Function;
```
Comment (aaronsn) [inline app/src/pages/ui/FipsSelector.tsx:14]: Can you give this a more concrete function type with args?

**Review by aaronsn — APPROVED** [review]
Very nice!

Unrelated to this PR, but I sort of wonder if sorting the whole county FIPS array and then creating a new FIPS class for each one of them could be a performance issue since it's happening on every render. Maybe we should just create the sorted Fips map once and then not need to change it every time.

**Comment by kkatzen** [comment]
> Very nice!
> 
> Unrelated to this PR, but I sort of wonder if sorting the whole county FIPS array and then creating a new FIPS class for each one of them could be a performance issue since it's happening on every render. Maybe we should just create the sorted Fips map once and then not need to change it every time.

yeah absolutely it's a bit of a mess haha

File: app/src/pages/ui/FipsSelector.tsx (hunk 1 above)
Comment (kkatzen) [reply app/src/pages/ui/FipsSelector.tsx:39]: oh lol duh xD

File: app/src/pages/ui/FipsSelector.tsx (hunk 2 above)
Comment (kkatzen) [reply app/src/pages/ui/FipsSelector.tsx:77]: This was on Chi's mocks... I'll  leave it and make a note to bring it up in next sync

**Pull request merged**